src/store_client.cc

```diff
@@ -527,12 +554,18 @@ store_client::readBody(const char *, ssize_t len)
              * copyInto.offset includes headers, which is what mem cache needs
              */
             int64_t mem_offset = entry->mem_obj->endOffset();
-            if ((copyInto.offset == mem_offset) || (parsed_header && mem_offset == rep->hdr_sz)) {
+            if ((copyInto.offset == mem_offset) || (headerParsed && mem_offset == rep->hdr_sz)) {
                 entry->mem_obj->write(StoreIOBuffer(len, copyInto.offset, copyInto.data));
             }
         }
     }
 
+    if (!headerParsed) {
+        // more data needed
+        copyInto.offset += len;
+        throw IncompleteHeaderException();
+    }
```

Comment:
Why do we increment copyInto.offset in this particular case only? What does that increment mean?